src/shims/backtrace.rs

```diff
@@ -1,29 +1,49 @@
 use crate::*;
 use rustc_ast::ast::Mutability;
 use rustc_middle::ty::layout::LayoutOf as _;
-use rustc_middle::ty::{self, TypeAndMut};
-use rustc_span::{BytePos, Symbol};
+use rustc_middle::ty::{self, Instance, TypeAndMut};
+use rustc_span::{BytePos, Loc, Symbol};
 use rustc_target::{abi::Size, spec::abi::Abi};
 use std::convert::TryInto as _;
 
 impl<'mir, 'tcx: 'mir> EvalContextExt<'mir, 'tcx> for crate::MiriEvalContext<'mir, 'tcx> {}
 pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> {
-    fn handle_miri_get_backtrace(
+    fn handle_miri_backtrace_size(
         &mut self,
         abi: Abi,
         link_name: Symbol,
         args: &[OpTy<'tcx, Tag>],
         dest: &PlaceTy<'tcx, Tag>,
     ) -> InterpResult<'tcx> {
         let this = self.eval_context_mut();
-        let tcx = this.tcx;
         let &[ref flags] = this.check_shim(abi, Abi::Rust, link_name, args)?;
 
         let flags = this.read_scalar(flags)?.to_u64()?;
-        if flags != 0 {
-            throw_unsup_format!("unknown `miri_get_backtrace` flags {}", flags);
+        if flags != 1 {
+            throw_unsup_format!("unknown `miri_backtrace_size` flags {}", flags);
```

Comment:
Why would we use any of these flag bits already?

This is not a version number, it's a bit field. At least that's how I have been thinking of it.